### PR TITLE
nextcloud container - allowed clients - only limit access in known use cases

### PR DESCRIPTION
**Pull request by szaimen**
Fix #3540

This is best reviewed like this: https://github.com/nextcloud/all-in-one/pull/3574/files?w=1

### CODE REVIEW

**Review by Zoey2936 — CHANGES_REQUESTED** [review]
that would make blocking other ips useless, since it is very easy to connect over the gateway to a container

File: Containers/nextcloud/start.sh [hunk 1]
```diff
@@ -135,15 +135,20 @@ while [ -z "$(dig nextcloud-aio-apache A +short)" ]; do
...
-IPv4_ADDRESS_MASTERCONTAINER="$(dig nextcloud-aio-mastercontainer A +short | grep '^[0-9.]\+$' | sort | head -n1)"
-IPv6_ADDRESS_MASTERCONTAINER="$(dig nextcloud-aio-mastercontainer AAAA +short | grep '^[0-9a-f:]\+$' | sort | head -n1)"
 
-sed -i "s|^;listen.allowed_clients|listen.allowed_clients|" /usr/local/etc/php-fpm.d/www.conf
-sed -i "s|listen.allowed_clients.*|listen.allowed_clients = 127.0.0.1,::1,$IPv4_ADDRESS_APACHE,$IPv6_ADDRESS_APACHE,$IPv4_ADDRESS_MASTERCONTAINER,$IPv6_ADDRESS_MASTERCONTAINER|" /usr/local/etc/php-fpm.d/www.conf
-sed -i "/^listen.allowed_clients/s/,,/,/g" /usr/local/etc/php-fpm.d/www.conf
-sed -i "/^listen.allowed_clients/s/,$//" /usr/local/etc/php-fpm.d/www.conf
-grep listen.allowed_clients /usr/local/etc/php-fpm.d/www.conf
+set -x
+if [ "$APACHE_PORT" = 443 ] || [ "$APACHE_IP_BINDING" = "127.0.0.1" ] || [ "$APACHE_IP_BINDING" = "::1" ]; then
```
Comment (szaimen) [inline Containers/nextcloud/start.sh:140]: @Zoey2936 this is most important. I decided based on your comment to only limit it if AIO runs in normal mode or the reverse proxy runs in the host network of the server that runs AIO. For these recommended cases we limit the access. Does this make sense?

**Comment by szaimen** [comment]
@Zoey2936 if you are fine with this PR, I would like to merge this and create a new beta release :)

**Review by Zoey2936 — APPROVED** [review]
we can try it

**Pull request merged**